### PR TITLE
[12.0][IMP] account_journal_lock_date: two dates per journal

### PR DESCRIPTION
**Pull request by ernestotejeda**
Based on https://github.com/OCA/account-financial-tools/pull/831 these changes were made:

- In journal model, change the existing 'Lock date' by two dates, the same as in company.
- Add a wizard to allows a massive update of those dates for several journals at the same time.

This video shows how an **Adviser** can set lock dates for a Journal and also for several journals at the same time using a wizard.
It also shows how an adviser cannot validate an invoice for a date before or equal to the 'Lock Date' defined in the 'Customer Invoices' journal, but can validate it for a later date, even if it is before the 'Lock Date for Non-Advisers' defined in the journal.
Adviser

This video shows how a **Non-Adviser** user cannot validate an invoice for a date before or equal to the 'Lock Date' or the 'Lock Date for Non-Advisers' defined in the 'Customer invoices' journal.
Non-adviser

cc @Tecnativa TT25028

### CODE REVIEW

File: account_journal_lock_date/README.rst [hunk 1]
```diff
@@ -27,26 +27,42 @@ Account Journal Lock Date
...
+#. Open a Journal and set the 'Lock Date' and the 'Lock Date for Non-Advisers'
+   or select several Journals in the list view and click on the action menu
+   'Update journals lock dates' to update those dates for the selected
+   journals at the same time.
+
+Usage
+=====
+
+If the logged-in user has the access group 'Adviser', he/she will
+not be able to create an journal entry if the 'Lock Date' of the
```
Comment (pedrobaeza) [inline account_journal_lock_date/README.rst]:
```suggestion
not be able to create a journal entry if the 'Lock Date' of the
```

File: account_journal_lock_date/README.rst [hunk 2]
```diff
@@ -27,26 +27,42 @@ Account Journal Lock Date
...
+journal is greater than or equal to the journal entry.
+
+If the logged-in user has not the access group 'Adviser', he/she will
+not be able to create an journal entry if the 'Lock Date for Non-Advisers'
```
Comment (pedrobaeza) [inline account_journal_lock_date/README.rst]:
```suggestion
not be able to create a journal entry if the 'Lock Date for Non-Advisers'
```

File: account_journal_lock_date/models/account_journal.py [hunk 3]
```diff
@@ -8,15 +8,16 @@ class AccountJournal(models.Model):
...
+    )
+    fiscalyear_lock_date = fields.Date(
+        string="Lock Date",
+        help="No users, including Advisers, can edit accounts prior "
+             "to and inclusive of this date for this journal. Use it "
+             "for fiscal year locking for this journal, for example."
     )
-
-    @api.model
-    def _can_bypass_journal_lock_date(self):
```
Comment (pedrobaeza) [inline account_journal_lock_date/models/account_journal.py]: Keep this method for inheritability

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please move the dates to page "Advanced settings" inside the group "Control-Access".

You can remove the draft state in the PR.

**Comment by ernestotejeda** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You haven't modified the position of the fields where I said, and you need to modify the README accordingly.

**Review by pedrobaeza — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Review by sergiocorato — APPROVED** [review]

**Pull request merged**